File: SpringBoot-Backend/mybank.account.transactions/src/main/java/com/mybank/account/transactions/service/AccountService.java
package com.mybank.account.transactions.service;

import com.mybank.account.transactions.model.AccountDetails;
import com.mybank.account.transactions.model.AccountTransactionsRpt;

import java.util.Date;

public interface AccountService {
    public AccountDetails getMonthlyBalance(String aAccountNumber, String asOfDate) throws Exception;
    public AccountDetails getCumulativeBalance(String aAccountNumber, String fromDate) throws Exception;
}

File: react-frontend/src/service/AccountService.js
import AccountApi from "../api/AccountApi";

export default class AccountService {

  static getMonthlyBalance(accountNumber, asOfDate) {
    return AccountApi({
      url: "/api/account/getmonthlybalance/" + accountNumber + "/" + asOfDate,
      method: "GET"
    });
  }
  static getCumulativeBalance(accountNumber, fromDate) {
    return AccountApi({
      url: "/api/account/getcumulativebalance/" + accountNumber + "/" + fromDate,
      method: "GET"
    });
  }
}
